src/Compilers/CSharp/Portable/Binder/Binder_Statements.cs

```diff
@@ -3636,6 +3640,251 @@ internal virtual BoundExpressionStatement BindConstructorInitializer(Constructor
             return constructorInitializer;
         }
 
+#nullable enable
+        internal BoundExpressionStatement? BindImplicitConstructorInitializer(SyntaxNode ctorSyntax, BindingDiagnosticBag diagnostics)
+        {
+            BoundExpression? initializerInvocation;
+            initializerInvocation = BindImplicitConstructorInitializer((MethodSymbol)this.ContainingMember(), diagnostics, Compilation);
+
+            if (initializerInvocation is null)
+            {
+                return null;
+            }
+
+            //  Base WasCompilerGenerated state off of whether constructor is implicitly declared, this will ensure proper instrumentation.
+            var constructorInitializer = new BoundExpressionStatement(ctorSyntax, initializerInvocation) { WasCompilerGenerated = ((MethodSymbol)ContainingMember()).IsImplicitlyDeclared };
```

Comment:
>WasCompilerGenerated [](http://example.com/codeflow?start=107&length=20)

It looks like this is almost never set to true here. How do we manage to show it as IsImplicit in IOperation?

Reply:
I was concerned with this myself, and spent some amount of time attempting to make `WasCompilerGenerated` always true, but the operation factory for `BoundExpressionStatement` already handles this case: https://github.com/dotnet/roslyn/blob/main/src/Compilers/CSharp/Portable/Operations/CSharpOperationFactory.cs#L2069. The underlying call and the expression statement share the same syntax node, so this will be fine. If you prefer, I could add some asserts to verify that this condition is always true.